BPCompatibleAlertControllerDemo/CompatibleAlertControllerDemo/ViewController.swift

```diff
@@ -9,52 +9,69 @@
 import UIKit
 
 class ViewController: UIViewController {
-    var alertController: BPCompatibleAlertController?
 
     override func viewDidLoad() {
         super.viewDidLoad()
         
+        var observerObject: NSObjectProtocol?
+        
+        let cleanupAfterAlertDismissed: (Void) -> Void =
```

Comment:
Hmmm. My feeling is that the need for using observers in the text field is particular to this demo's use case, rather than a core responsibility of the alert controller in general use cases. (None of the use-cases in my code bases actually use them.) 

To integrate removing observers in the alert controller, the alert controller would also need to be responsible for calling `addObserverForName` as the returned object from that call is needed to pass to `removeObserver`. It then needs to 'know' quite a bit more about a specific use case and might be hard to generalise...

To ease the use case where observers are required, potentially the alert controller could have a user configurable property something like `var postActionHandlerCleanupFunction: ((Void) -> Void)?` that it calls automatically at the end of `postAlertDismissalActions()`. Then the demo code wouldn't need to manually remember to call `cleanupAfterAlertDismissed()` in each handler block. Would something like this be helpful?